sdk/eventhub/Azure.Messaging.EventHubs/src/Core/ChannelEnumerableSubscription.cs

```diff
@@ -59,10 +53,11 @@ internal class ChannelEnumerableSubscription<T> : IAsyncEnumerable<T>, IAsyncDis
                 Guard.ArgumentNotNegative(nameof(maximumWaitTime), maximumWaitTime.Value);
             }
 
-            ChannelReader = reader;
-            MaximumWaitTime = maximumWaitTime;
             UnsubscribeCallbackAsync = unsubscribeCallbackAsync;
-            CancellationToken = cancellationToken;
+
+            SourceEnumerator = new Lazy<IAsyncEnumerator<T>>(
```

Comment:
Why do you use lazy? AFAIK all calls to IAsyncEnumerable are sequential and not coming from different threads.

Reply:
I wanted to avoid creating the enumerator unless it was actually used, since it's disposable and calls into `EnumerateChannel`.   Since the enumerable is now also the enumerator, you could potentially direct cast to `IAsyncEnumerator` and start using it without going through `GetAsyncEnumerator`, so my initial thought of putting it in there seemed less appealing.

Think it would be better to just directly initialize in the constructor and assume that it'll be used under normal circumstances?

Reply:
.... and for the historical record, my understanding was incorrect here.  The execution was deferred not just in the actual iteration but also with creation of the enumerable/enumerator.  The instantiation would not have run until the enumerator's `MoveNext` was invoked.   

Thus, Pavel is absolutely correct in that Lazy was entirely unnecessary and this could have been set in the constructor directly with no resource concerns.